File: lib/utilities/transition/parameter-stack.ts
import * as cdk from "aws-cdk-lib";
import * as ssm from "aws-cdk-lib/aws-ssm";

import { Construct } from "constructs";

import previousPersistentStorageStack from "./.env";

export class TransitionParameterStack extends cdk.Stack {
  readonly snapshotLambdaNameParameter: ssm.StringParameter;
  readonly previousSnapshotIdParameter: ssm.StringParameter;

  constructor(scope: Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    this.snapshotLambdaNameParameter = new ssm.StringParameter(
      this,
      "SnapshotLambdaNameParameter",
      { stringValue: "TBD" }
    );

    this.previousSnapshotIdParameter = new ssm.StringParameter(
      this,
      "PreviousSnapshotIdParameter",
      { stringValue: previousPersistentStorageStack.serverStorageSnapshotId }
    );
  }
}

File: nitecap/utilities/transition/parameter_stack.py
import aws_cdk as cdk
import aws_cdk.aws_ssm as ssm

from constructs import Construct
from ...configuration import Configuration


class TransitionParameterStack(cdk.Stack):
    def __init__(
        self,
        scope: Construct,
        construct_id: str,
        *,
        configuration: Configuration,
        **kwargs,
    ) -> None:
        super().__init__(scope, construct_id, **kwargs)

        self.snapshot_lambda_name_parameter = ssm.StringParameter(
            self, "SnapshotLambdaNameParameter", string_value="TBD"
        )

        self.previous_snapshot_id_parameter = ssm.StringParameter(
            self,
            "PreviousSnapshotIdParameter",
            string_value=configuration.previous_server_storage_snapshot_id,
        )
